src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,645 @@
+                            {
+                                if (TryFindParentAssignment(source, out var parentStatement))
+                                {
+                                    var variableDeclaration =
+                                        SyntaxFactory.LocalDeclarationStatement(
+                                            SyntaxFactory.VariableDeclaration(
+                                                VarNameIdentifier,
+                                                SyntaxFactory.SingletonSeparatedList(
+                                                    SyntaxFactory.VariableDeclarator(
+                                                        identifierName.Identifier,
+                                                        null,
+                                                        SyntaxFactory.EqualsValueClause(initializer)))));
+                                    var returnList = SyntaxFactory.ReturnStatement(identifierName);
+                                    documentUpdate = new DocumentUpdate(parentStatement, new[] { variableDeclaration, qe2, returnList });
+                                    return true;
+                                }
+                            }
+
+                            break;
+                        case SyntaxKind.ArrowExpressionClause:
+                            // Not supported
+                            break;
+                        default:
+                            throw new NotImplementedException(parenthesizedExpression.Parent.Parent.Parent.Kind().ToString());
+                    }
+                    documentUpdate = default;
+                    return false;
+                }
+
+                throw new NotImplementedException(parenthesizedExpression.Parent.Parent.Kind().ToString());
+            }
+
+            private bool TryGetVariableName(SyntaxNode node, out ExpressionSyntax identifier)
+            {
+                switch (node.Kind())
+                {
+                    case SyntaxKind.EqualsValueClause:
+                        identifier = SyntaxFactory.IdentifierName(((VariableDeclaratorSyntax)node.Parent).Identifier);
+                        return true;
+                    case SyntaxKind.MultiplyAssignmentExpression:
+                    case SyntaxKind.SimpleAssignmentExpression:
+                        identifier = ((AssignmentExpressionSyntax)node).Left;
+                        return true;
+                }
+
+                identifier = null;
+                return false;
+            }
+
+            private bool TryReplaceWithLocalFunction(QueryExpressionSyntax source, out DocumentUpdate documentUpdate)
+            {
+                if (TryFindParentAssignment(source, out var parentStatement))
+                {
+                    string localFunctionName = GetFreeSymbolName("localFunction", source.GetLocation().SourceSpan.Start);
```

Comment:
var

Reply:
the string type is not evident here. prefer to define it explicitly.

---
In reply to: [175267747](https://github.com/dotnet/roslyn/pull/25362#discussion_r175267747) [](ancestors = 175267747)